geode-core/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterQueryServiceCommandWithSecurityDUnitTest.java

```diff
@@ -61,81 +61,119 @@
-
-  private static MemberVM locator;
-  private static List<MemberVM> servers = new ArrayList<>();
+  public ClusterStartupRule cluster = new ClusterStartupRule();
 
   @Before
   public void setUp() throws Exception {
     locator = cluster.startLocatorVM(0, l -> l.withSecurityManager(SimpleSecurityManager.class));
     int locatorPort = locator.getPort();
-    IntStream.range(0, serversToStart).forEach(
-        i -> servers.add(i, cluster.startServerVM(i + 1, s -> s.withConnectionToLocator(locatorPort)
-            .withCredential("clusterManage", "clusterManage"))));
+    IntStream.range(0, 2).forEach(i -> servers.add(i, cluster.startServerVM(i + 1, s -> s
+        .withConnectionToLocator(locatorPort).withCredential("clusterManage", "clusterManage"))));
     gfsh.connectAndVerify(locator);
   }
 
+  private String buildCommand(String authorizerName, String authorizerParameters) {
+    StringBuilder commandBuilder = new StringBuilder(COMMAND_NAME);
+    commandBuilder.append(" --").append(AUTHORIZER_NAME).append("=").append(authorizerName);
+
+    if (authorizerParameters != null) {
+      commandBuilder.append(" --").append(AUTHORIZER_PARAMETERS).append("=")
+          .append(authorizerParameters);
+    }
+
+    return commandBuilder.toString();
+  }
+
+  private void executeAndAssertGfshCommandIsSuccessful(Class authorizerClass,
+      String authorizerParameters) {
+    gfsh.executeAndAssertThat(buildCommand(authorizerClass.getName(), authorizerParameters))
+        .statusIsSuccess().hasTableSection().hasRowSize(servers.size());
+
+    verifyCurrentAuthorizerClass(authorizerClass);
+  }
+
+  private void startServerAndVerifyJavaBeanAccessorMethodAuthorizer() {
+    int locatorPort = locator.getPort();
+    MemberVM newServer = cluster.startServerVM(servers.size() + 1, s -> s
+        .withConnectionToLocator(locatorPort).withCredential("clusterManage", "clusterManage"));
+    newServer.invoke(() -> {
+      JavaBeanAccessorMethodAuthorizer methodAuthorizer =
+          (JavaBeanAccessorMethodAuthorizer) Objects.requireNonNull(ClusterStartupRule.getCache())
+              .getService(QueryConfigurationService.class).getMethodAuthorizer();
+      assertThat(methodAuthorizer.getClass()).isEqualTo(JavaBeanAccessorMethodAuthorizer.class);
+      Set<String> expectedParameters =
+          new HashSet<>(Arrays.asList(JAVA_BEAN_AUTHORIZER_PARAMETERS.split(SPLITTING_REGEX)));
+      assertThat(methodAuthorizer.getAllowedPackages()).isEqualTo(expectedParameters);
+    });
+  }
+
+  private void verifyCurrentAuthorizerClass(Class authorizerClass) {
+    servers.forEach(server -> server.invoke(() -> {
+      MethodInvocationAuthorizer methodAuthorizer =
+          Objects.requireNonNull(ClusterStartupRule.getCache())
+              .getService(QueryConfigurationService.class).getMethodAuthorizer();
+      assertThat(methodAuthorizer.getClass()).isEqualTo(authorizerClass);
+    }));
+  }
+
+  private String getFilePath() throws IOException {
```

Comment:
Since this method now always gets the file path for the `TestMethodAuthorizer.txt` file, it might be better to rename it something like "getTestMethodAuthorizerFilePath" to make it clearer what it's doing when called.

Reply:
Sounds good, done!.